apache/ambari
# AMBARI-23230 Upgrade Solr configuration while upgrading Ambari 2.6.1 …

…-> 2.7.0

## What changes were proposed in this pull request?

- upgrade solrconfig.xml of logsearch, solr, ranger and atlas stored in the Ambari database while upgrading Ambari
- workaround for upgrading logsearch-server and logfeeder: rpm and deb packages does not create links in the /etc/ambari-logesearch-[logfeeder|portal] directory
- set the default conf directory to /usr/lib/ambari-logsearch[logfeeder|portal]/conf
- fix: logsearch_ui_protocol and logsearch_ui_port values where exchanged during Ambari upgrade

## How was this patch tested?

Unit tests for regular expressions
Manually using vagrant:
- Install Ambari 2.6.2 and deploy a HDP 2.6 cluster: zookeeper, logsearch, ambari-infra, ranger  
- Stop logsearch
- Delete collections from Infra Solr
- upgrade Ambari to 2.7.0
- upgrade Infra Solr and Logsearch in each host manually.
- Start Logsearch
- Check all collections are created
- Check all the solrconfig.xmls are upgraded 

Please review:
@oleewere @zeroflag @g-boros
